lib/src/type_helper.dart

```diff
@@ -2,25 +2,27 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'package:analyzer/dart/element/element.dart';
 import 'package:analyzer/dart/element/type.dart';
-
 import 'package:source_gen/source_gen.dart' show TypeChecker;
 
-abstract class TypeHelper {
-  const TypeHelper();
+/// If [type] is the [Type] or implements the [Type] represented by [checker],
+/// returns the generic arguments to the [checker] [Type] if there are any.
+///
+/// If the [checker] [Type] doesn't have generic arguments, `null` is returned.
+List<DartType> getTypeArguments(DartType type, TypeChecker checker) {
```

Comment:
[nit] could be `typeArguments` or `typeArgumentsOf`... `get` doesn't add value most of the time.